src/training.py

```diff
@@ -269,21 +270,62 @@ def run_experiment(
         - test_pred_dist: The test prediction distribution
         - mcc: The test Matthews correlation coefficient
     """
+
+    trains = []
+    target_train = None
+    target_valid = None
+    target_test = None
     th.manual_seed(1984)
 
-    train_data, valid_data, test_data = create_datasets(
-        symbol,
-        seq_len=seq_len,
-        fixed_scaling=[(7, 3000.), (8, 12.), (9, 31.)],
-        log_splits=log_splits,
-        root=f"{root}/data/clean"
+    for symbol in train_symbols:
+        train_data, valid_data, test_data = create_datasets(
+            symbol,
+            seq_len=seq_len,
+            fixed_scaling=[(7, 3000.), (8, 12.), (9, 31.)],
+            log_splits=log_splits,
+            root=f"{root}/data/clean"
+        )
+
+        trains.append(train_data)
+        if symbol == target_symbol:
+            target_train = train_data
+            target_valid = valid_data
+            target_test = test_data
+
+    concatted_trains = ConcatDataset(trains)
+
+    train_loader = DataLoader(concatted_trains, batch_size=batch_size, shuffle=True)
+    valid_loader = DataLoader(target_valid, batch_size=batch_size, shuffle=False)
+    test_loader = DataLoader(target_test, batch_size=batch_size, shuffle=False)
+
+
+    train_label_ct = th.sum(th.stack([x.target_counts for x in trains]),dim=0)
+    #train_label_ct = concatted_trains.target_counts
+
+    epochs = trainer_params['epochs']
+    trainer_params['epochs'] = int(split*epochs)
+
+
+    pretrain = train_model(
+        model=None,
+        x_dim=train_data[0][0].shape[1],
+        train_loader=train_loader,
+        valid_loader=valid_loader,
+        test_loader=test_loader,
+        train_label_ct=train_label_ct,
+        model_class=model,
+        model_params=model_params, 
+        trainer_params=trainer_params
     )
-    train_loader = DataLoader(train_data, batch_size=batch_size, shuffle=True)
-    valid_loader = DataLoader(valid_data, batch_size=batch_size, shuffle=False)
-    test_loader = DataLoader(test_data, batch_size=batch_size, shuffle=False)
-    train_label_ct = train_data.target_counts
 
+    trainer_params['epochs'] = int((1-split)*epochs)
+    train_loader = DataLoader(target_train, batch_size=batch_size, shuffle=True)
+    train_label_ct = target_train.target_counts
+    trainer_params['lr'] = trainer_params['lr'] * trainer_params['fine_tune_lr_ratio']
```

Comment:
Are we expecting every trainer to have the fine_tune_lr_ratio? either this needs to be treated as optional or all trainer configs need this added.